nidirect_taxonomy_navigator/src/TaxonomyAdminBreadcrumb.php

```diff
@@ -0,0 +1,161 @@
+   *   The menu link access service.
+   * @param \Symfony\Component\Routing\Matcher\RequestMatcherInterface $router
+   *   The dynamic router service.
+   * @param \Drupal\Core\PathProcessor\InboundPathProcessorInterface $path_processor
+   *   The inbound path processor.
+   * @param \Drupal\Core\Config\ConfigFactoryInterface $config_factory
+   *   The config factory service.
+   * @param \Drupal\Core\Controller\TitleResolverInterface $title_resolver
+   *   The title resolver service.
+   * @param \Drupal\Core\Session\AccountInterface $current_user
+   *   The current user object.
+   * @param \Drupal\Core\Path\CurrentPathStack $current_path
+   *   The current path.
+   * @param \Drupal\Core\Path\PathMatcherInterface $path_matcher
+   *   The path matcher service.
+   * @param array $route_matches
+   *   Matching routes injected from service parameters.
+   */
+  public function __construct(RequestContext $context, AccessManagerInterface $access_manager, RequestMatcherInterface $router, InboundPathProcessorInterface $path_processor, ConfigFactoryInterface $config_factory, TitleResolverInterface $title_resolver, AccountInterface $current_user, CurrentPathStack $current_path, PathMatcherInterface $path_matcher = NULL, array $route_matches) {
+    parent::__construct($context, $access_manager, $router, $path_processor, $config_factory, $title_resolver, $current_user, $current_path, $path_matcher);
+    $this->routeMatches = $route_matches;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function applies(RouteMatchInterface $route_match) {
+    $match = FALSE;
+
+    // Check for matching route names.
+    if (in_array($route_match->getRouteName(), $this->routeMatches)) {
+      $match = TRUE;
+    }
+
+    return $match;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build(RouteMatchInterface $route_match) {
+    $breadcrumb = new Breadcrumb();
+    $links = [];
+
+    // General path-based breadcrumbs. Use the actual request path, prior to
+    // resolving path aliases, so the breadcrumb can be defined by simply
+    // creating a hierarchy of path aliases.
+    $path = trim($this->context->getPathInfo(), '/');
+    $path_elements = explode('/', $path);
+    $exclude = [];
+
+    // Intercept/replace the core taxonomy route with our taxonomy navigator equivalent.
+    foreach ($path_elements as $index => $element) {
+      if ($element == 'taxonomy') {
+        $path_elements[$index] = 'taxonomy_navigator';
+      }
+
+      // Drop the redundant manage path element (from core).
+      if ($element == 'manage') {
+        unset($path_elements[$index]);
+      }
+    }
+
+    while (count($path_elements) > 1) {
+      array_pop($path_elements);
+
+      // Copy the path elements for up-casting.
+      $route_request = $this->getRequestForPath('/' . implode('/', $path_elements), $exclude);
+
+      if ($route_request) {
+        $route_match = RouteMatch::createFromRequest($route_request);
```

Comment:
Possibly inject the current route path?

Reply:
Yep, agreed! Funnily enough this was lifted from core... unclear why it wasn't passed in there too.

Reply:
Ah, it's a global static function... that'll be why then! https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Routing%21RouteMatch.php/function/RouteMatch%3A%3AcreateFromRequest/8.2.x

Reply:
Ahh that explains it then, good find.